buildchain/buildchain/targets/remote_image.py

```diff
@@ -77,54 +73,94 @@ def fullname(self) -> str:
     def task(self) -> types.TaskDict:
         task = self.basic_task
         task.update({
-            'title': self._show,
+            'title': lambda _: self.show('PULL IMG'),
             'doc': 'Download {} container image.'.format(self.name),
+            'targets': [self.dirname/'manifest.json'],
             'actions': self._build_actions(),
             'uptodate': [True],
+            'clean': [self.clean],
         })
         return task
 
-    @staticmethod
-    def _show(task: types.Task) -> str:
-        """Return a description of the task."""
-        return utils.title_with_target1('IMG PULL', task)
-
     def _build_actions(self) -> List[types.Action]:
-        """Compute actions for the image — pull, tag, save."""
-        filepath = self.uncompressed_filename
-        pull_callable = docker_command.DockerPull(
-            repository=self.repository, digest=self.digest
-        )
-        # The local repository is the image 'tag' without version
-        local_repository = self.tag[:self.tag.rfind(':')]
-        tag_callable = docker_command.DockerTag(
-            repository=local_repository,
-            full_name=self.fullname,
-            version=self.version
-        )
-        save_callable = docker_command.DockerSave(
-            tag=self.tag, save_path=filepath
-        )
-
-        actions : List[types.Action] = [
-            pull_callable,
-            tag_callable,
-            save_callable
+        return [
+            self.mkdirs,
+            [
+                config.SKOPEO, 'copy',
+                '--format', 'v2s2',
+                '--dest-compress',
+                'docker://{}'.format(self.fullname),
+                'dir:{}'.format(str(self.dirname))
+            ]
         ]
-        # containerd doesn't support compressed images.
-        if not self.for_containerd:
-            actions.append((coreutils.gzip, [filepath], {}))
-        return actions
+
+
+class RemoteTarImage(base.FileTarget):
```

Comment:
So, not handling #609 in this PR?

Reply:
I tried to, using `skopeo copy … docker-archive:path/to/image` + using `ctr -n k8s.io image import path/to/image` but bootstrap wasn't working. I'll look at the logs tomorrow, to see if it's easy to fix or not.

Reply:
Done now :)